### PR TITLE
Use log_artifact in notebook.

### LINKED ISSUE #171: experiments: Include model in the experimetns inside notebook
In the current notebook, we are creating multiple experiments inside a cell by using `save_dvc_exp=True` inside a for loop:

```python
for base_lr in [0.001, 0.005, 0.01]:
    with Live(str(ROOT / "results" / "train"), save_dvc_exp=True) as live:
        live.log_param("train_arch", train_arch)
        fine_tune_args = {
            'epochs': 8,
            'base_lr': base_lr
        }
        live.log_params(fine_tune_args)

        learn = unet_learner(data_loader, 
                            arch=getattr(models, train_arch), 
                            metrics=DiceMulti)
        learn.fine_tune(
            **fine_tune_args,
            cbs=[DVCLiveCallback(live=live)])
```

Although metrics, params, and plots are correctly saved for each experiment, the model is basically lost and the `learn` instance we are working with is just the one from the last iteration.

We should serialize the model and include it in each experiment.

This requires https://github.com/iterative/dvclive/issues/378 and is just an example of when it would be useful to have.

### PR DESCRIPTION
**Pull request by daavoo**
Closes #171 

It is currently deployed:

Captura de pantalla 2023-04-13 a las 18 02 51

### CODE REVIEW

**Review by daavoo — COMMENTED** [review]
Actually found a bug in experiment refs in Studio (doesn't display the details of the `.dvc` file inside the exp ref)

File: example-get-started-experiments/generate.sh [hunk 1]
```diff
@@ -95,6 +95,7 @@ cp -r $HERE/code/src .
 cp $HERE/code/params.yaml .
 sed -e "s/base_lr: 0.01/base_lr: $BEST_EXP_BASE_LR/" -i".bkp" params.yaml
 rm params.yaml.bkp
+dvc remove models/model.pkl.dvc
```
[the remark this answers is not in the document]
Comment (daavoo) [reply example-get-started-experiments/generate.sh:98]: We transition from `log_artifact` which is `dvc add` to stage output in the `dvc.yaml`.

[the remark this answers is not in the document]
Comment (shcheklein) [reply example-get-started-experiments/generate.sh:98]: got it ... and what it was before the `log_artifact` where and how did we save the model?

[the remark this answers is not in the document]
Comment (daavoo) [reply example-get-started-experiments/generate.sh:98]: > what it was before the log_artifact where and how did we save the model?

We didn't save it at all during the `notebook` state, that was the motivation for the P.R.

[the remark this answers is not in the document]
Comment (shcheklein) [reply example-get-started-experiments/generate.sh:98]: got it!

**Review by shcheklein — APPROVED** [review]

**Review by dberenbaum — COMMENTED** [review]
Thanks @daavoo! We should also add `log_artifact` to `train.py`, but it's not a blocker if we want to wait until we are ready to add `type=model` or other MR stuff since it should be a noop now. Do we have that tracked somewhere?

Minor non-blocking thought: the DVCLive code is starting to look a little complex. Should we add some comments to explain what's happening and make clear that you can pick and choose which calls you need?

**Comment by daavoo** [comment]
> We should also add log_artifact to train.py, but it's not a blocker if we want to wait until we are ready to add type=model or other MR stuff since it should be a noop now. Do we have that tracked somewhere?

Um. So the model would not be a stage output?

**Comment by dberenbaum** [comment]
No, the model can still be a stage output, that's why it would be a noop now and I think it's okay to not include it yet. Once `log_artifact` also adds it to the `artifacts` section, we will want it inside `train.py` also. I think it also may confuse users if `log_artifact` is used in the notebook but not `train.py`.

**Pull request merged**